tests/framework/framework.go

```diff
@@ -22,3 +31,89 @@ var (
 	// IntegrationTestRunner runs etcdserver.EtcdServer in separate goroutine and uses client libraries to communicate.
 	IntegrationTestRunner = integrationRunner{}
 )
+
+const TickDuration = 10 * time.Millisecond
+
+// WaitLeader returns index of the member in c.Members() that is leader
+// or fails the test (if members don't agree on a legal leader in 30s).
+func WaitLeader(t testing.TB, c Cluster) int {
```

Comment:
This  is against the existing design, it is supposed to be method in the `Cluster` interface instead of a function.

Reply:
Yeah, this is the bad side. Removing `WaitLeader` from `Cluster` interface would make it a little counterintuitive.